packages-web/maps/src/Maps.editorConfig.ts

```diff
@@ -0,0 +1,220 @@
+        const properties = defaultPropertiesForDynamicMarkers?.[index];
+        if (f.locationType === "address") {
+            hideProperty<DynamicMarkersType>("latitude", properties);
+            hideProperty<DynamicMarkersType>("longitude", properties);
+        } else {
+            hideProperty<DynamicMarkersType>("address", properties);
+        }
+        if (f.markerStyleDynamic === "default") {
+            hideProperty<DynamicMarkersType>("customMarkerDynamic", properties);
+        }
+    });
+
+    if (values.mapProvider !== "googleMaps") {
+        hideProperty<MapsPreviewProps>("optionStreetView", defaultProperties);
+        hideProperty<MapsPreviewProps>("mapTypeControl", defaultProperties);
+        hideProperty<MapsPreviewProps>("fullScreenControl", defaultProperties);
+        hideProperty<MapsPreviewProps>("rotateControl", defaultProperties);
+        hideProperty<MapsPreviewProps>("mapStyles", defaultProperties);
+        if (values.mapProvider === "openStreet") {
+            hideProperty<MapsPreviewProps>("apiKey", defaultProperties);
+        }
+    } else {
+        hideProperty<MapsPreviewProps>("attributionControl", defaultProperties);
+    }
+
+    return defaultProperties;
+}
+
+export function check(values: MapsPreviewProps): Problem[] {
+    const errors: Problem[] = [];
+    values.markers
+        .filter(marker => values.advanced && marker.markerStyle === "image" && !marker.customMarker)
+        .forEach(marker => {
+            errors.push({
+                property: "customMarker",
+                severity: "error",
+                message: `Custom marker image is required when shape is 'image' for address ${marker.address ??
+                    marker.addressAttribute ??
+                    marker.addressExpression}`
+            });
+        });
+    if (values.mapProvider !== "openStreet" && !values.apiKey) {
+        errors.push({
+            property: "apiKey",
+            severity: "error",
+            message: "To avoid errors during map rendering it's necessary to include an Api Key",
+            url: "https://github.com/mendix/widgets-resources/blob/master/packages-web/maps/README.md#limitations"
+        });
+    } else if (values.advanced && !values.geodecodeApiKey) {
+        errors.push({
+            property: "geodecodeApiKey",
+            severity: "warning",
+            message: "To translate addresses to latitude and longitude a Google API Key is required",
+            url: "https://github.com/mendix/widgets-resources/blob/master/packages-web/maps/README.md#limitations"
+        });
+    }
+
+    values.markers.forEach(marker => {
+        if (marker.dataSourceType === "static") {
+            if (marker.locationType === "address") {
+                if (!marker.address) {
+                    errors.push({
+                        property: "markers.address",
+                        severity: "error",
+                        message: "A static marker requires an address"
```

Comment:
Can we extract the error object to a function and spit out the errors ?
Then in 10 places you can do
`!marker.address && errors.push(givemeError("address"))`

Reply:
I removed the other props